src/control/server/ctl_ranks_rpc.go

```diff
@@ -295,30 +296,47 @@ func (svc *ControlService) ResetFormatRanks(ctx context.Context, req *ctlpb.Rank
 	}
 	svc.log.Debugf("CtlSvc.ResetFormatRanks dispatch, req:%+v\n", req)
 
-	instances, err := svc.harness.FilterInstancesByRankSet(req.GetRanks())
+	engineInstances, err := svc.harness.FilterInstancesByRankSet(req.GetRanks())
 	if err != nil {
 		return nil, err
 	}
 
+	nvmeScanResp, err := svc.NvmeScan(storage.BdevScanRequest{})
+	if err != nil {
+		svc.log.Errorf("nvme scan failed: %s", err)
+		nvmeScanResp = &storage.BdevScanResponse{}
+	}
+	if nvmeScanResp == nil {
```

Comment:
Do not see, how nil could be returned without an error.
I am probably missing something?
Maybe to support future backend provider?